src/lib.rs

```diff
@@ -19,55 +50,282 @@ pub mod common {
     include!(concat!(env!("OUT_DIR"), "/common.rs"));
 }
 
-use common::MavMessage;
-
-const MAV_STX: u8 = 0xFE;
+/// Encapsulation of all possible Mavlink messages
+pub use self::common::MavMessage as MavMessage;
 
 /// Metadata from a MAVLink packet header
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub struct Header {
-    pub sequence: u8,
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Default)]
+pub struct MavHeader {
     pub system_id: u8,
     pub component_id: u8,
+    pub sequence: u8,
+}
+
+
+/// Message framing marker for mavlink v1
+pub const MAV_STX: u8 = 0xFE;
+
+/// Message framing marker for mavlink v2
+pub const MAV_STX_V2: u8 = 0xFD;
+
+
+impl MavHeader {
+    /// Return a default GCS header, seq is replaced by the connector
+    /// so it can be ignored. Set `component_id` to your desired component ID.
+    pub fn get_default_header() -> MavHeader {
+        MavHeader {
+            system_id: 255,
+            component_id: 0,
+            sequence: 0,
+        }
+    }
+}
+
+/// Encapsulation of the Mavlink message and the header,
+/// important to preserve information about the sender system
+/// and component id
+#[derive(Debug, Clone)]
+pub struct MavFrame {
+    pub header: MavHeader,
+    pub msg: MavMessage,
+}
+
+impl MavFrame {
+    /// Create a new frame with given message
+    pub fn new(msg: MavMessage) -> MavFrame {
+        MavFrame {
+            header: MavHeader::get_default_header(),
+            msg
+        }
+    }
+
+    /// Serialize frame into a vector, so it can be send
+    /// over a socket for example
+    pub fn ser(&self) -> Vec<u8> {
+        let mut v = vec![];
+
+        // serialize header
+        v.push(self.header.system_id);
+        v.push(self.header.component_id);
+        v.push(self.header.sequence);
+
+        // message id
+        #[cfg(feature="mavlink2")]
+        {
+            let bytes: [u8; 4] = unsafe { transmute(self.msg.message_id().to_le()) };
```

Comment:
```suggestion
            let bytes: [u8; 4] = self.msg.message_id().to_le_bytes();
```
(newly stabilized in Rust 1.32)